cmd/docker-app/deploy.go

```diff
@@ -51,10 +55,33 @@ func deployCmd(dockerCli command.Cli) *cobra.Command {
 	cmd.Flags().BoolVarP(&opts.deploySendRegistryAuth, "with-registry-auth", "", false, "Sends registry auth")
 	if internal.Experimental == "on" {
 		cmd.Flags().StringArrayVarP(&opts.deployComposeFiles, "compose-files", "c", []string{}, "Override Compose files")
+		cmd.Flags().BoolVarP(&opts.deployNoRenderConfig, "disable-render-config", "", false, "Do not apply renderer to configuration files")
 	}
 	return cmd
 }
 
+func renderAttachments(app *types.App, tempDir string, d map[string]string) error {
+	for _, a := range app.Attachments() {
+		content, err := ioutil.ReadFile(filepath.Join(app.Path, a.Path()))
+		if err != nil {
+			return err
+		}
+		rendered, err := render.RenderConfig(app, d, string(content))
+		if err != nil {
+			return errors.Wrapf(err, "failed to render %s", a.Path())
+		}
+		err = os.MkdirAll(filepath.Dir(filepath.Join(tempDir, a.Path())), 0700)
+		if err != nil {
+			return errors.Wrap(err, "failed to create temporary directory")
+		}
+		err = ioutil.WriteFile(filepath.Join(tempDir, a.Path()), []byte(rendered), 0600)
+		if err != nil {
```

Comment:
There's a bunch of opportunities for the `if err := ... ; err != nil {` pattern to be used in this block (the final three out of the four cases I think). Other than that this LGTM (FWIW).

Reply:
Not a big fan of the pattern especially when the lines are already very long.

Reply:
Yeah, Go idiom doesn't seem to worry about line lengths so much as you or I might.

In this case pulling `filepath.Join(tempDir, a.Path()` (used twice) into a variable would probably mitigate it enough to make it ok though.